cli/flox/src/commands/activate.rs

```diff
@@ -297,47 +298,51 @@ impl Activate {
                 .to_string(),
             );
 
-            // TODO: we should clean up the different conditionals here
             if in_place && self.start_services {
                 debug!("not starting services for in-place activation");
                 message::warning("Skipped starting services. Services are not yet supported for in place activations.");
             }
+            // We should error for remote environments even if they don't have
+            // services so that the user doesn't assume we're actually starting
+            // services.
+            if self.start_services {
+                supported_environment(&flox, &self.environment)?; // Error for remote envs.
+            }
             if flox.features.services && !manifest.services.is_empty() && !in_place {
-                if self.start_services {
-                    supported_environment(&flox, &self.environment)?; // Error for remote envs.
-                }
-                tracing::debug!(
-                    start = self.start_services,
-                    socket_exists = socket_path.exists(),
-                    "setting service variables"
+                // Always set the socket var because we use it to register
+                // cleanup in tests
+                // We can probably drop this extra conditional once we drop
+                // `register_cleanup.sh`
+                exports.insert(
+                    FLOX_SERVICES_SOCKET_VAR,
+                    socket_path.to_string_lossy().to_string(),
                 );
-                if self.start_services && socket_path.exists() {
-                    debug!("detected existing services socket");
-                    debug!("will not start services");
-                    message::warning("Skipped starting services, services are already running");
-                } else {
-                    if self.start_services {
-                        debug!("will start services");
+                if self.start_services {
+                    let start_new_process_compose = if socket_path.exists() {
+                        shutdown_process_compose_if_all_processes_stopped(&socket_path)?
```

Comment:
(non-blocking observation) I've realised that this has a potentially surprising effect if all of your services are short-lived commands that run to completion because each `activate --start-services` will no longer be idempotent and will restart them again. Not having any long running services seems like a strange use of services and we shouldn't worry about the edge case.

Reply:
We could add a check for whether the environment has actually changed. I'm not sure if that's necessarily desirable though

Reply:
> Not having any long running services seems like a strange use of services and we shouldn't worry about the edge case.

I dont think having only short living services is too far from an edge case.

Or more generally, i think it would be a feature to start ("run") short lived services ("scripts") that go along with the manifest.

Currently the only way to do that is to define aliases/functions or well, services (in which case they will be backgrounded and probably at least not run with `--start-services`.

Its probably more of an enhancement/feature kind that we can ignore for now as to not inflate services with functionality they are not intended for

Reply:
They smell like `hooks` 😄